common/src/cellular_pktio.c

```diff
@@ -547,7 +515,7 @@ static char * _Cellular_ReadLine( CellularContext_t * pContext,
     char * pRead = NULL;  /* pRead is the first empty ptr in the Buffer for comm intf to read. */
     uint32_t bytesRead = 0;
     uint32_t partialDataRead = pContext->partialDataRcvdLen;
-    uint32_t bufferEmptyLength = PKTIO_READ_BUFFER_SIZE;
+    int32_t bufferEmptyLength = PKTIO_READ_BUFFER_SIZE;
```

Comment:
Because the `else condition` in the following code will not be true if the data type is uint32_t
```
    if( bufferEmptyLength > 0 )
    {
        ( void ) pContext->pCommIntf->recv( pContext->hPktioCommIntf, ( uint8_t * ) pRead,
                                            bufferEmptyLength,
                                            CELLULAR_COMM_IF_RECV_TIMEOUT_MS, &bytesRead );

        if( bytesRead > 0U )
        {
            /* Add a NULL after the bytesRead. This is required for further processing. */
            pRead[ bytesRead ] = '\0';

            CellularLogDebug( "AT Read %d bytes, data[%p]", bytesRead, pRead );
            /* Set the pBytesRead only when actual bytes read from comm interface. */
            *pBytesRead = bytesRead + partialDataRead;

            /* Clean the partial data and read pointer. */
            pContext->partialDataRcvdLen = 0;
        }
        else
        {
            pAtBuf = NULL;
            *pBytesRead = 0U;
        }
    }
    else
    {
        CellularLogError( "No empty space from comm if to handle incoming data, reset all parameter for next incoming data." );
        *pBytesRead = 0;
        pContext->partialDataRcvdLen = 0;
        pContext->pPktioReadPtr = NULL;
    }
```

Reply:
This fix consider the negative value situation, but int32_t is used to store the uint32_t operation result.
Suggest we do more check to prevent int32_t overflow in MISRA check.

Reply:
Sure, I think there should be other work for MISRA check.